src/routes/dashboard/user/contests/contests.remote.ts

```
@@ -0,0 +1,36 @@
+import { query } from '$app/server';
+import { createContestService } from '$lib/services/ContestService';
+import { error } from '@sveltejs/kit';
+import { getRequestEvent } from '$app/server';
+import type { UserContestsResponse } from '$lib/dto/contest';
+
+export const getUserContests = query(async (): Promise<UserContestsResponse> => {
+  const { cookies, locals } = getRequestEvent();
+
+  if (!locals.user) {
+    throw error(401, 'Unauthorized');
+  }
+
+  try {
+    const contestService = createContestService(cookies);
+    const contests = await contestService.getUserContests(locals.user.userId);
+
+    // Ensure we return a valid UserContestsResponse structure
+    if (!contests || typeof contests !== 'object') {
+      return {
+        ongoing: [],
+        upcoming: [],
+        past: []
+      };
+    }
```

we should throw an error like in different queries, not play as nothing happens